### PR TITLE
Maybe need to add conventions about plugins?

### PR DESCRIPTION
**Pull request by patsuckow**
Maybe need to add conventions about plugins? in conventions.rst
I'm not an expert, but it's necessary to add this to the agreements ? (but I can be wrong).

### CODE REVIEW

**Comment by dereuromark** [comment]
This is already mentioned at https://github.com/FriendsOfCake/awesome-cakephp/blob/master/CONTRIBUTING.md#tips-for-creating-cakephp-plugins
We could either link them (to keep things DRY) or outline the important points.
I agree that a few conventions are helpful.

**Comment by saeideng** [comment]
:-1: 
plugin name is optional

**Comment by dereuromark** [comment]
@saeideng You need to understand the difference between requirement and convention/guideline.
That's why he used SHOULD here.

File: en/intro/conventions.rst [hunk 1]
```diff
@@ -156,6 +156,13 @@ Now that you've been introduced to CakePHP's fundamentals, you might try a run
 through the :doc:`/tutorials-and-examples/cms/installation` to see how things fit
 together.
 
+Plugins Conventions
+===================
+The names of the plugins must be semantically significant.
```
Comment (dereuromark) [inline en/intro/conventions.rst]: Can you rephrase to e.g.

> It is useful to prefix a CakePHP plugin with "cakephp-" in the package name. This makes the name semantically related on the framework it depends on.

**Comment by markstory** [comment]
I think recommending conventions is useful to help promote consistency, and having it both here and in the awesome list is not a bad thing.

**Comment by saeideng** [comment]
>You need to understand the difference between requirement and convention/guideline.

yes I know
but Cake uses these conventions to work/understand what you want to do 
my understanding of cake conventions is : 
when you coding with cake conventions , you no need to config all thins 
and Cake will do for you 

but for plugins name , cake not uses / prefer conventions , at least what things are in this PR

**Comment by saeideng** [comment]
if you need these 
I prefer to having a section as `recommendations`

File: en/intro/conventions.rst [hunk 2]
```diff
@@ -156,6 +156,14 @@ Now that you've been introduced to CakePHP's fundamentals, you might try a run
 through the :doc:`/tutorials-and-examples/cms/installation` to see how things fit
 together.
 
+Plugins Conventions
+===================
+It is useful to prefix a CakePHP plugin with "cakephp-" in the package name.
+This makes the name semantically related on the framework it depends on.
+Ideally, this should be a prefix dependencies of 'cakephp' as the basis.
```
Comment (dereuromark) [inline en/intro/conventions.rst]: Now you got this duplicated.

File: en/intro/conventions.rst [hunk 3]
```diff
@@ -156,6 +156,14 @@ Now that you've been introduced to CakePHP's fundamentals, you might try a run
...
+Do **not** use the CakePHP namespace (cakephp) as this is reserved to CakePHP owned plugins.
```
Comment (dereuromark) [inline en/intro/conventions.rst]: I would rather link to the awesome list conventions, as you are duplicating the content here now isolated from that list.

Comment (patsuckow) [reply en/intro/conventions.rst]: I'm already a little confused. Please, can you edit it yourself the way you see it right? Or write completely here in the commentary so that I make a complete change. If it is necessary at all, because I see that not everyone agrees and I do not quite understand what to do next.

Comment (saeideng) [reply en/intro/conventions.rst]: >because I see that not everyone agrees 

my mean of above text is 
this is not a cake convention but is cake recommendation
and if @dereuromark  was agree , we can change `Plugins Conventions` to something like `Plugins Recommendation`

Comment (dereuromark) [reply en/intro/conventions.rst]: Isn't a convention a recommendation? Just mildly more opinionated (for good reason)?

**Comment by lorenzo** [comment]
Can you add a couple examples after the new paragraph showing good and bad plugin names?

**Comment by patsuckow** [comment]
@lorenzo, idea to add a recommendation appeared on the basis of documentation - https://book.cakephp.org/3.0/en/plugins.html#publish-your-plugin , there is all the description, but the fact is that I'm not sure that I can correctly describe the entire text of this recommendation. So I immediately warned that I'm not an expert. Therefore, I asked @dereuromark to propose a version of this recommendation.

**Comment by dereuromark** [comment]
I added a link on top and an example.

**Comment by patsuckow** [comment]
Thank you @dereuromark  :)
